packages/valory/skills/market_manager_abci/behaviours.py

```diff
@@ -138,6 +141,12 @@ def _update_bets(
         self,
     ) -> Generator:
         """Fetch the questions from all the prediction markets and update the local copy of the bets."""
+
+        # Extract the last queue number from the bets
+        # This is used to determine the next queue number
+        if self.bets:
+            self.current_queue_number = max(bet.queue_no for bet in self.bets) + 1
```

Comment:
This increases the queue's number in each period! Instead, it should be increasing when it is fully consumed.